### PR TITLE
added functionality for bookUpdate

### PR DESCRIPTION
**Pull request by gcater**
The landing page index.tsx now only has a bookForm component. On a successful book create the user will get sent to a new page "src/pages/book/[bookid]/index.tsx". This page has a BookUpdate Component and the BookRenderer component. User can make changes to the book info or add a chapter.

book.ts got updateBook mutation and getBook query. BookRenderer component now uses getBook which requires a bookId.

src/pages/book/[bookid]/addChapter.tsx and src/pages/book/[bookid]/[chapterid]/addSection.tsx now send BookRenderer a bookId var.

### CODE REVIEW

File: src/components/BookUpdate.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    .string({
+      required_error: "Author is required.",
+    })
+    .min(2, {
+      message: "Author must be at least 2 characters.",
+    }),
+});
+
+const BookUpdate = ({ bookId }: { bookId: string }): JSX.Element => {
+  const { data: queryData, refetch: refetchBook } =
```
Comment (Evanc123) [inline src/components/BookUpdate.tsx]: Rename

File: src/components/BookUpdate.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+  const { mutate: updateBook } = api.book.updateBook.useMutation({
+    onSuccess: (data) => {
+      console.log(data);
+      void refetchBook();
+    },
+  });
+  const { mutate: deleteBook } = api.book.deleteBook.useMutation({
+    onSuccess: () => {
+      window.location.href = "/";
```
Comment (Evanc123) [inline src/components/BookUpdate.tsx]: Fix

File: src/components/BookUpdate.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            onSubmit={handleUpdate}
+          >
+            <Button type="submit">Update Book</Button>
+          </AutoForm>
+        </CardContent>
+        {bookId && (
+          <>
+            <CardContent>
+              <Button>
+                <a href={`/book/${bookId}/addChapter`}>Add Chapter</a>
```
Comment (Evanc123) [inline src/components/BookUpdate.tsx]: Never use a href in a next application, always use next link <Link />

**Review by Evanc123 — CHANGES_REQUESTED** [review]

**Comment by gcater** [comment]
Changed filename UpdateBook.tsx to EditBook.tsx. No longer using href, instead using router.push.

File: src/components/BookUpdate.tsx (hunk 3 above)
Comment (gcater) [reply src/components/BookUpdate.tsx]: <a onClick={() => router.push(`/book/${bookId}/addChapter`)}>
 
 Would this be a correct alternative?

File: src/components/BookUpdate.tsx (hunk 1 above)
Comment (gcater) [reply src/components/BookUpdate.tsx]: Is this because there are other functions basically named the same thing?

Comment (Evanc123) [reply src/components/BookUpdate.tsx]: Query data is just unclear

File: src/components/BookUpdate.tsx (hunk 3 above)
Comment (Evanc123) [reply src/components/BookUpdate.tsx]: That would be okay, but next link is better

Comment (Evanc123) [reply src/components/BookUpdate.tsx]: https://nextjs.org/docs/pages/api-reference/components/link 

This is the primary way, but if hou need to do it programmatically you y can use router push (but in this case it's just a button)

**Comment by gcater** [comment]
Changed EditBook back to BookUpdate, renamed "querydata" to "updated book", changed the "a hrefs" to "link hrefs" in BookForm and ChapterForm. Made the BookForm take up the whole screen in pages/index.tsx

File: src/components/BookUpdate.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    .string({
+      required_error: "Author is required.",
+    })
+    .min(2, {
+      message: "Author must be at least 2 characters.",
+    }),
+});
+
+const BookUpdate = ({ bookId }: { bookId: string }): JSX.Element => {
+  const { data: updatedBook, refetch: refetchBook } =
```
Comment (Evanc123) [inline src/components/BookUpdate.tsx:34]: yo ucan just call it `book`

File: src/pages/index.tsx [hunk 5]
```diff
@@ -18,11 +17,7 @@ export default function Home() {
             <AuthShowcase />
           </div>
         </div>
-
-        <div className="flex w-full">
-          <div className="w-1/2">{BookForm()}</div>
-          <div className="w-1/2">{BookRenderer()}</div>
-        </div>
+        <div className="w-1/2">{BookForm()}</div>
```
Comment (Evanc123) [inline src/pages/index.tsx:20]: i'd rather have this be a component than a function, but we can address later

**Review by Evanc123 — APPROVED** [review]

**Pull request merged**